### ISSUE TITLE
SP800-56A rev3 FFC failed to generate vectors

### ISSUE THREAD

**Issue by smuellerDD** [usnistgov/ACVP#828]
**environment**
Demo

**testSessionId**
80542

**vsId**
186456

**Algorithm registration**
```
[
  {
    "acvVersion":"1.0"
  },
  {
    "isSample":false,
    "operation":"register",
    "certificateRequest":"no",
    "debugRequest":"yes",
    "production":"no",
    "encryptAtRest":"yes",
    "algorithms":[
      {
        "prereqVals":[
          {
            "algorithm":"SHA",
            "valValue":"same"
          },
          {
            "algorithm":"HMAC",
            "valValue":"same"
          },
          {
            "algorithm":"DRBG",
            "valValue":"same"
          },
          {
            "algorithm":"SafePrimes",
            "valValue":"same"
          },
          {
            "algorithm":"SP800-108",
            "valValue":"same"
          }
        ],
        "algorithm":"KAS-FFC",
        "revision":"Sp800-56Ar3",
        "function":[
          "keyPairGen",
          "partialVal",
          "fullVal"
        ],
        "iutId":"1234567890abcdef",
        "scheme":{
          "MQV1":{
            "kasRole":[
              "initiator",
              "responder"
            ],
            "kdfMethods":{
              "oneStepKdf":{
                "auxFunctions":[
                  {
                    "auxFunctionName":"HMAC-SHA2-256",
                    "macSaltMethods":[
                      "default",
                      "random"
                    ]
                  },
                  {
                    "auxFunctionName":"SHA2-256",
                    "macSaltMethods":[
                      "random"
                    ]
                  }
                ],
                "fixedInfoPattern":"uPartyInfo||vPartyInfo||literal[affedeadbeef]",
                "encoding":[
                  "concatenation"
                ]
              },
              "twoStepKdf":{
                "capabilities":[
                  {
                    "macSaltMethods":[
                      "default",
                      "random"
                    ],
                    "fixedInfoPattern":"uPartyInfo||vPartyInfo||literal[0123456789abcdef]",
                    "encoding":[
                      "concatenation"
                    ],
                    "kdfMode":"counter",
                    "macMode":[
                      "HMAC-SHA-1",
                      "HMAC-SHA2-224",
                      "HMAC-SHA2-256",
                      "HMAC-SHA2-384",
                      "HMAC-SHA2-512"
                    ],
                    "supportedLengths":[
                      {
                        "min":8,
                        "max":4096,
                        "increment":8
                      }
                    ],
                    "fixedDataOrder":[
                      "before fixed data"
                    ],
                    "counterLength":[
                      32
                    ],
                    "supportsEmptyIv":false,
                    "requiresEmptyIv":false
                  },
                  {
                    "macSaltMethods":[
                      "default"
                    ],
                    "fixedInfoPattern":"literal[fedcba9876543210]||context||algorithmId",
                    "encoding":[
                      "concatenation"
                    ],
                    "kdfMode":"feedback",
                    "macMode":[
                      "HMAC-SHA-1",
                      "HMAC-SHA2-224",
                      "HMAC-SHA2-256",
                      "HMAC-SHA2-384",
                      "HMAC-SHA2-512"
                    ],
                    "supportedLengths":[
                      {
                        "min":8,
                        "max":4096,
                        "increment":8
                      }
                    ],
                    "fixedDataOrder":[
                      "before fixed data"
                    ],
                    "counterLength":[
                      32
                    ],
                    "supportsEmptyIv":false,
                    "requiresEmptyIv":false
                  }
                ]
              }
            },
            "keyConfirmationMethod":{
              "macMethods":{
                "HMAC-SHA2-256":{
                  "keyLen":128,
                  "macLen":128
                },
                "HMAC-SHA2-512":{
                  "keyLen":128,
                  "macLen":128
                }
              },
              "keyConfirmationDirections":[
                "unilateral",
                "bilateral"
              ],
              "keyConfirmationRoles":[
                "provider",
                "recipient"
              ]
            },
            "l":1024
          }
        },
        "domainParameterGenerationMethods":[
          "MODP-2048"
        ]
      }
    ]
  }
]
```

Returned error:

```
[ {
  "acvVersion" : "1.0"
}, {
  "vsId" : 186456,
  "status" : "error during vector generation",
  "message" : ""
} ]
```

PS: IFC (KAS and KTS) and ECC generation worked

**Comment by Kritner** [comment]
tagging #684, #685

**Comment by Kritner** [comment]
@smuellerDD I put in a PR to correct the labels, they were correct in the json sample information, but not in the description of the schemes.

TLDR: `MQV1` -> `mqv1`

**Comment by smuellerDD** [comment]
Am Dienstag, 14. April 2020, 13:37:32 CEST schrieb Russ Hammett:

Hi Russ,

> @smuellerDD I put in a PR to correct the labels, they were correct in the
> json sample information, but not in the description of the schemes.
> 
> TLDR: `MQV1` -> `mqv1`

Confirmed, working

Ciao
Stephan
